tests/feature_elimination/test_feature_elimination.py

```diff
@@ -72,6 +73,58 @@ def test_shap_rfe(X, y, capsys):
     out, _ = capsys.readouterr()
     assert len(out) == 0
 
+def test_shap_rfe_cols_to_keep(X, y, capsys):
+    """
+    Test for shap_rfe_cv with feautures to keep parameter.
+    """
+    clf = DecisionTreeClassifier(max_depth=1,random_state=1)
+    with pytest.warns(None) as record:
+        shap_elimination = ShapRFECV(clf, random_state=1, step=1, cv=2, scoring='roc_auc', n_jobs=4,min_features_to_select=1)
+        shap_elimination = shap_elimination.fit(X, y,columns_to_keep=['col_2'])
+
+    assert shap_elimination.fitted == True
+    shap_elimination._check_if_fitted()
+
+    report = shap_elimination.compute()
+
+    assert report.shape[0] == 3
+    reduced_feature_set = set(shap_elimination.get_reduced_features_set(num_features=1))
+    assert reduced_feature_set == set(['col_2'])
+
+    # Ensure that number of warnings was 0
+    assert len(record) == 0
+    # Check if there is any prints
+    out, _ = capsys.readouterr()
+    assert len(out) == 0
+
+def test_shap_rfe_randomized_search_cols_to_keep(X, y, capsys):
+    
+    clf = DecisionTreeClassifier(max_depth=1)
+    param_grid = {
+        'criterion': ['gini'],
+        'min_samples_split': [1, 2]
+    }
+    search = RandomizedSearchCV(clf, param_grid, cv=2, n_iter=2)
+    with pytest.warns(None) as record:
+
+        shap_elimination = ShapRFECV(search, step=0.8, cv=2, scoring='roc_auc', n_jobs=4,random_state=1)
+        report = shap_elimination.fit_compute(X, y,columns_to_keep=['col_2'])
```

Comment:
In this case lets try to have 2 cols to keep, minimum number of features at 1 and step =2. This would be an "extreme" case

Reply:
Done. The tests pass for this case.